### PR TITLE
Library project

### PR DESCRIPTION
**Pull request by hillarioh**
Hi, 
This is a request for code review of the javascript library project. It entailed 
- creating a book object to store data for the Book read or not read by the user.
- functions that generate the HTML for the books.
- Local Storage, so the info persists

Thanks in advance

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        </div>
+        <div class="form-group">
+          <label for="author">Author: </label>
+          <input
+            type="text"
+            class="form-control"
+            name="author"
+            id="author"
+            placeholder="Enter author name"
+          />
```
Comment (mosaaleb) [inline index.html:42]: - [ ] The form should have a `number of pages`, `status` fields

File: js/index.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  this.title = document.querySelector('#booktitle').value;
+  this.author = document.querySelector('#author').value;
+  this.read = false;
+}
+
+// get books from localstorage
+const getBooks = () => {
+  const lib = localStorage.getItem('library');
+  return lib ? JSON.parse(lib) : [];
+};
```
Comment (mosaaleb) [inline js/index.js:14]: - [ ] Let's reset form after successful submit. 
  You can use [`reset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reset) on form.

File: js/index.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  let booksList = '';
+  books.forEach((book, i) => {
+    const { title, author, read, pages } = book;
+    booksList += `
+      <tr>
+        <th scope="row">${i + 1}</th>
+        <td>${title}</td>
+        <td>by <i>${author}</i></td>
+        <td>${pages ? pages : 'unknown'}</td>
+        <td> ${read}</td>
```
Comment (fatymahmed) [inline js/index.js:28]: - Each Book displayed should have a button to toggle the Book `read` status, see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/toggle_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library) - in this case I should be able to change the status when the book is being displayed.

File: js/index.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
+const btn = document.querySelector('#newbook');
+const bookForm = document.querySelector('#bookform');
+
+function Book() {
+  this.title = document.querySelector('#booktitle').value;
+  this.author = document.querySelector('#author').value;
+  this.read = document.querySelector('#read').checked;
+  this.pages = document.querySelector('#pages').value;
+}
```
Comment (fatymahmed) [inline js/index.js:9]: [OPTIONAL]
- Your app should have an array to store the instances of Book, please see the [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/book_array_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library) which has more information on how to use the array
**Even though this is marked as optional, I highly recommend you to work on this**

File: README.md [hunk 5]
```diff
@@ -1 +1,66 @@
...
+- Local Storage, so the info persists through the refresh of the page.
+
+## Built With
+
+- HTML 5 and CSS 3.
+- Javascript
+- Bootstrap
+
+## Live Demo
+
```
Comment (fatymahmed) [inline README.md:25]: [OPTIONAL]
- Add a live link to your readme file

**Pull request merged**